mlir/utils/performance/analysis/compileAndCollectTuningData.py

```diff
@@ -0,0 +1,795 @@
+
+# Global template for tuning data structure
+TUNING_DATA_TEMPLATE = {
+    'blocksize': None,
+    'gridsize': None,
+    'vgpr_count': None,
+    'vgpr_spills': None,
+    'sgpr_count': None,
+    'sgpr_spills': None,
+    'LDS_allocated': None,
+    'occupancy': None
+}
+
+def create_tuning_data():
+    return TUNING_DATA_TEMPLATE.copy()
+
+def check_rocmlir_binaries():
+    """
+    Check if rocmlir-gen and rocmlir-driver binaries can be found.
+    Returns the paths to the binaries if found, otherwise exits.
+    """
+    # Get the script's directory and find the rocMLIR root
+    script_dir = os.path.dirname(os.path.abspath(__file__))
+
+    # Navigate up from mlir/utils/performance to find rocMLIR root
+    rocmlir_root = os.path.abspath(os.path.join(script_dir, '..', '..', '..',
+                                                '..'))
+
+    # Define expected binary paths
+    build_bin_dir = os.path.join(rocmlir_root, 'build', 'bin')
+    build_external_bin_dir = os.path.join(rocmlir_root, 'build', 'external',
+                                          'llvm-project', 'llvm', 'bin')
+    rocmlir_gen_path = os.path.join(build_bin_dir, 'rocmlir-gen')
+    rocmlir_driver_path = os.path.join(build_bin_dir, 'rocmlir-driver')
+    rocmlir_translate_path = os.path.join(build_bin_dir, 'rocmlir-translate')
+    opt_path = os.path.join(build_external_bin_dir, 'opt')
+    llc_path = os.path.join(build_external_bin_dir, 'llc')
+
+    # Check if build/bin directory exists
+    if not os.path.exists(build_bin_dir):
+        print(f"Error: Build directory not found at {build_bin_dir}")
+        print("Please make sure the rocMLIR build directory exists.")
+        sys.exit(1)
+
+    # Check for rocmlir-gen
+    if not os.path.exists(rocmlir_gen_path):
+        print(f"Error: rocmlir-gen not found at {rocmlir_gen_path}")
+        sys.exit(1)
+
+    # Check for rocmlir-driver
+    if not os.path.exists(rocmlir_driver_path):
+        print(f"Error: rocmlir-driver not found at {rocmlir_driver_path}")
+        sys.exit(1)
+
+    # Check for rocmlir-translate
+    if not os.path.exists(rocmlir_translate_path):
+        print(f"Error: rocmlir-translate not found at {rocmlir_translate_path}")
+        sys.exit(1)
+
+    # Check for opt
+    if not os.path.exists(opt_path):
+        print(f"Error: opt not found at {opt_path}")
+        sys.exit(1)
+
+    # Check for llc
+    if not os.path.exists(llc_path):
+        print(f"Error: llc not found at {llc_path}")
+        sys.exit(1)
+
+    print(f"✓ Found rocmlir-gen: {rocmlir_gen_path}")
+    print(f"✓ Found rocmlir-driver: {rocmlir_driver_path}")
+    print(f"✓ Found rocmlir-translate: {rocmlir_translate_path}")
+    print(f"✓ Found opt: {opt_path}")
+    print(f"✓ Found llc: {llc_path}")
+
+    return [rocmlir_gen_path, rocmlir_driver_path, rocmlir_translate_path,
+           opt_path, llc_path]
+
+def parse_config_csv(config_file):
+    """Parse the input CSV file containing configuration data."""
```

Comment:
```suggestion
    """Parse the input tuning database debug file containing configuration data."""
```
It is `tsv` file instead of `csv`